junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -141,4 +143,78 @@ private static boolean hasParameters(Method method) {
 		}
 	}
 
+	/**
+	 * {@code DisplayNameGenerator} that generate complete sentences.
+	 *
+	 * <p>This extends the functionality of {@link ReplaceUnderscores} by
+	 * generating complete sentences display names, divided by a separator
+	 * {@link IndicativeSentencesGeneration}.
+	 *
+	 * @since 5.6
+	 */
+	@API(status = EXPERIMENTAL, since = "5.6")
+	class IndicativeSentences extends ReplaceUnderscores {
+
+		@Override
+		public String generateDisplayNameForClass(Class<?> testClass) {
+			return super.generateDisplayNameForClass(testClass).contains("$") ? testClass.getSimpleName()
+					: super.generateDisplayNameForClass(testClass);
+		}
+
+		@Override
+		public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+			return classReplaceToIndicativeSentence(nestedClass);
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			return classReplaceToIndicativeSentence(testClass) + getSentenceSeparator(testClass)
+					+ super.generateDisplayNameForMethod(testClass, testMethod);
+		}
+
+		private String classReplaceToIndicativeSentence(Class<?> testClass) {
+			Class<?> enclosingParent = testClass.getEnclosingClass();
+			Optional<DisplayName> displayName = AnnotationSupport.findAnnotation(testClass, DisplayName.class);
+			Optional<DisplayNameGeneration> displayNameGeneration = AnnotationSupport.findAnnotation(testClass,
+				DisplayNameGeneration.class);
+
+			if (enclosingParent == null) {
+				return displayName.map(DisplayName::value).orElseGet(() -> generateDisplayNameForClass(testClass));
+			}
+			else { //&& displayNameGeneration.getClass() == IndicativeSentences.class
```

Comment:
❓ Is this sth. you meant to check?

Reply:
Oh that was just a miss comment that i]I got left to remove, I was just playing around with those methods values while debugging, I'll get rid of it on the https://github.com/junit-team/junit5/pull/2124#discussion_r360117501 changes.